src/Lecture2.hs

```diff
@@ -58,7 +63,7 @@ lazyProduct = error "TODO"
 "ccaabb"
 -}
 duplicate :: [a] -> [a]
-duplicate = error "TODO"
+duplicate = concatMap (replicate 2)
```

Comment:
That's a correct solution 👍🏻 However, the existing implementation allocates temporary lists inside and appends them with `++`.

It is more efficient to construct lists using the cons operator (`:`) directly without introducing intermediate values or adding lists with `(++)` whenever possible.